crates/stc_ts_file_analyzer/src/analyzer/decl_merging.rs

```diff
@@ -97,24 +97,37 @@ impl Analyzer<'_, '_> {
             }
 
             (Type::Interface(a), Type::Interface(bi)) => {
-                // TODO: Handle the number of type parameters.
                 let mut type_params = FxHashMap::default();
-                if let Some(b_tps) = &bi.type_params {
-                    if let Some(a_tp) = &a.type_params {
+
+                match (&a.type_params, &bi.type_params) {
+                    (Some(a_tps), Some(b_tps)) => {
+                        if a_tps.params.len() != b_tps.params.len() {
+                            self.storage
+                                .report(ErrorKind::InterfaceNonIdenticalTypeParams { span: a.span() }.into());
```

Comment:
span currently looks like this
```rs
interface A<T extends string> {}
────────────────────────────────
```
is there any way to only use the span of the name of the interface? (like tsc)

```rs
```rs
interface A<T extends string> {}
          ─
```
```

Reply:
I think we should add a field like `name_span` to the `Interface` type